keps/sig-network/1880-services-ip-ranges/README.md

```diff
@@ -0,0 +1,935 @@
+- Change the default IPFamily used for Services IP allocation, the defaulting depends on the
+  services.spec.IPFamily and services.spec.IPFamilyPolicy, a simple webhook or an admission plugin
+  can set this fields to the desired default, so the allocation logic doesn't have to handle it.
+
+## Proposal
+
+The proposal is to implement a new allocator logic that consumes 2 new API Objects: ServiceIPRange
+and ServiceIP, and allow users to dynamically increase the number of Services IPs available by
+creating new ServiceIPRanges.
+
+The new allocator will be able to "automagically" consume IPs from any of this new ranges, we can
+think about this model, as the same as adding more disks to a Storage system to increase the
+capacity.
+
+To simplify the model, make it backwards compatible and to avoid that it can evolve into something
+different and collide with other APIs, like Gateway APIs, we are adding the following constraints:
+
+ - a ServiceIPRange will be immutable after creation
+ - a ServiceIPRange can only be deleted if there are no ServiceIP associated to it
+ - there can not be overlapping IP networks on the ServiceIPRanges created
+ - there has to be always an unique and default ServiceIPRange
+ - any ServiceIP existing in a cluster has to belong to a ServiceIPRange
+ - any Service with a ClusterIP assigned has to have always a ServiceIP object associated
+
+### User Stories
+
+#### Story 1
+
+As a Kubernetes user I want to be able to dynamically increase the number of IPs availables for
+Services.
+
+#### Story 2
+
+As a Kubernetes admin I want to have a process that allows me to renumber my Services IPs.
+
+#### Story 3
+
+As a Kubernetes developer I want to be able to know the current Service IP range and the IP
+addresses allocated.
+
+#### Story 4
+
+As a Kubernetes admin that wants to use IPv6, I want to be able to follow the IPv6 address planning
+best practices, being able to assign /64 prefixes to my end subnets.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+Current API machinery doesn't consider transactions, Services API is kind of special in this regard
+since already performs allocations inside the Registry pipeline using the Storage to keep
+consistency on an Obscure API Object that stores the bitmap with the allocations.
+
+This proposal also requires to perform validation from one Object against other Object inside an API
+request.
+
+### Risks and Mitigations
+
+Cross validation of Objects is not common in the Kubernetes API, sig-apimachinery should verify not
+bad precedent is introduced.
+
+Current Kubernetes cluster have a very static network configuration, allowing to expand the Service
+IP ranges will give more flexibility to users, with the risk of having problematic or incongruent
+network configurations with overlapping. But this is not really a new problem, users need to do a
+proper network planning before adding new ranges to the Service IP pool.
+
```

Comment:
added a paragraph, they consume the clusterIP field, unless they were doing some assumptions about the range, it should not be any problem.
If they were considering the range, it had to be a configuration parameter at cluster creation, since there is no way a component can obtain the service range parameters, they can now use the new IPRange object to obtain this information